ctapipe/reco/sklearn.py

```diff
@@ -898,6 +905,29 @@ def __init__(self, model_component, **kwargs):
                 "Unsupported Model of type %s supplied", self.model_component
             )
 
+        if self.output_path:
```

Comment:
You can combine these two into `if self.output_path and self.output_path.exists():` without any danger, if `output_path` is none the second bit is not executed.

Reply:
But if I do it like this, the lines below the checks for the correct configuration regarding `overwrite` do not get executed if the output path doesn't already exist. So I would have to put those in a separate if statement which would result in the same number of if statements as right now.